modules/doi/includes/utilities.inc

```diff
@@ -864,3 +864,32 @@ function islandora_doi_get_mods($id) {
 
   return islandora_doi_translate($source_string, $content_type);
 }
+
+/**
+ * Strip HTML tags from imported XML metadata.
+ *
+ * @param DOMDocument $mods_domdoc
+ *   The DOMDocument representing the MODS record being built
+ *
+ * @param DOMDocument $doi_domdoc
+ *   DOMDocument representing the DOI metadata returned from Crossref
+ *
+ * @param DOMElement $doi_child_node
+ *   DOMElement representing a node of the $doi_domdoc being iterated through
+ *
+ * @param string $element_name
+ *   The name of the element being stripped and rebuilt
+ *
+ * @return DOMElement 
+ *   DOMElement newly created for inclusion in $mods_domdoc
+ */
+function islandora_doi_create_html_stripped_element($mods_domdoc, $doi_domdoc, $doi_child_node, $element_name) {
+  $element_text = $doi_domdoc->saveXML($doi_child_node);
+  $element_text = str_replace("<{$element_name}>", '', $element_text);
+  $element_text = str_replace("</{$element_name}>", '', $element_text);
+  $element_text = str_replace('<', '&lt;', $element_text);
```

Comment:
Manual manipulation of structured data can lead to a world of issues. I recommend using the available XML parser.

Reply:
@willtp87 I'm not sure I understand what you mean

Reply:
Probably to replace the initial `->saveXML()` with `foreach($doi_child_node as $element){}` ? (though, I'm testing this at the moment, and struggling with the newlines that Crossref uses in their XML. I thought DOMDocument would have a 'no pretty formatting' option, and I've tried two, and am still struggling to get them to work. The `preg_replace()` below might be the way to do it, even though regex adds extra processing time. We don't want/need newlines in our metadata do we?  :D (UUUGH this is gonna bite us in the ass if we have abstracts, I've been here before)

Reply:
@rosiel I'm all for stripping newlines, and would think `preg_replace()` would be the way to do it. As for moving from manual string formatting to manipulation via SimpleXML or DOMDocument, I suppose instead of deleting the wrapping element the way we are currently doing, we could just get the content of the first child and then treat THAT internal bit as a string?

Reply:
@bryjbrown  regarding  https://github.com/Islandora/islandora_scholar/pull/288/files#r189376164
please consider
```
$doc = new DOMDocument;
$elem = $doc->createElement(
'my_elem', 
'<a b="blah_blah">_some_string_</a>'
);
$doc->appendChild($elem);
dsm($doc->saveXML());
``` 
outputing
```
<?xml version="1.0"?>
<my_elem>&lt;a b="blah_blah"&gt;_some_string_&lt;/a&gt;</my_elem>
```
No manual string manipulation was used.  When creating text nodes XML parsers properly encode content. If one relies on the libraries provided then they do not need to account for all the difficult edge cases as others have already done so.